programs/management/commands/import_all_program_configs.py

```
@@ -134,49 +221,82 @@ def _handle_dry_run(self, pending_files: list[Path]) -> None:
         results = ImportResults()
 
+        # A --only pass applies part of a config, so it must not stamp a hash covering the
+        # whole file — recording a partial apply as complete is the same defect as recording
+        # a skip. Reconcile ignores tracking state when selecting files, so a partial pass
+        # loses nothing by recording nothing.
+        results.partial_pass = bool(only) and set(only) != set(RECONCILE_SECTIONS)
+
+        verb = "reconcile" if reconcile else "import"
         self.stdout.write(self.style.WARNING(f"\n{'=' * 60}"))
+        if dry_run:
+            self.stdout.write(self.style.WARNING(f"DRY RUN - No changes will be made"))
```

Ruff (0.16.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)